@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -620,99 +584,90 @@ export class TileGeometryCreator {
+                    //         );
+                    //     });
+                    // }
+
+                    if (material === undefined) {
+                        continue;
+                    }
+                    materials[techniqueIndex] = material;
                 }
 
-                const bufferGeometry = new THREE.BufferGeometry();
-
-                srcGeometry.vertexAttributes.forEach((vertexAttribute: BufferAttribute) => {
-                    const buffer = getBufferAttribute(vertexAttribute);
-                    bufferGeometry.setAttribute(vertexAttribute.name, buffer);
-                });
-
-                if (srcGeometry.interleavedVertexAttributes !== undefined) {
-                    srcGeometry.interleavedVertexAttributes.forEach(
-                        (attr: {
-                            type: any;
-                            buffer: any;
-                            stride: any;
-                            attributes: {
-                                forEach: (
-                                    arg0: (interleavedAttr: {
-                                        itemSize: any;
-                                        offset: any;
-                                        name: any;
-                                    }) => void
-                                ) => void;
-                            };
-                        }) => {
-                            const ArrayCtor = getArrayConstructor(attr.type);
-                            const buffer = new THREE.InterleavedBuffer(
-                                new ArrayCtor(attr.buffer),
-                                attr.stride
-                            );
-                            attr.attributes.forEach(
-                                (interleavedAttr: { itemSize: any; offset: any; name: any }) => {
-                                    const attribute = new THREE.InterleavedBufferAttribute(
-                                        buffer,
-                                        interleavedAttr.itemSize,
-                                        interleavedAttr.offset,
-                                        false
-                                    );
-                                    bufferGeometry.setAttribute(interleavedAttr.name, attribute);
-                                }
-                            );
-                        }
+                if (hasTechniqueHandler) {
+                    const techniqueHandler = tile.techniqueHandlerPool.getTechniqueHandler(
+                        technique as IndexedTechnique,
+                        tile,
+                        mapView.techniqueUpdateContext
                     );
-                }
 
-                if (srcGeometry.index) {
-                    bufferGeometry.setIndex(getBufferAttribute(srcGeometry.index));
-                }
+                    // TODO: text technique handler and world space technique handler doesn't
```

Comment:
Maybe instead of splitting the technique handler into `WorldTechniqueHandler` and `ScreenTechniqueHandler` we could just have both methods:

- `createWorldObjects` and
- `createScreenObjects` in the interface.

For the current techniques one of them would just return empyt buffers, but maybe in the future we could even have a technique that produces geometry and labels at the same time.